lib/nucleus_rails/generator.rb

```diff
@@ -0,0 +1,277 @@
+# rubocop:disable Metrics/ClassLength, Metrics/MethodLength
+class NucleusRails::Generator < Rails::Generators::NamedBase
+  source_root(File.expand_path("app", __dir__))
+
+  desc "Creates necessary files to setup REST endpoints for the given database tables"
+  def create_rest_endpoints_from_database_tables(tables=[])
+    fetch_table_names(tables).each do |name|
+      generate_files_for(name)
+    end
+  end
+
+  private
+
+  def fetch_table_names(table_names=[])
+    tables = ActiveRecord::Base.connection.tables.map(&:downcase).to_set
+    tables = tables.intersection(table_names) if table_names.any?
+    tables = tables.delete('schema_migrations')
+
+    return tables
+  end
+
+  def generate_files_for(name)
```

Comment:
All generated classes should inherit from a NucleusRails super class, not NucleusCore